lib/platform-stats.js

```diff
@@ -99,30 +100,135 @@ export const updateDailyStationStats = async (
 
 /**
  * @param {import('pg').Client} pgClient
- * @param {Set<string>} participants
+ * @param {number[]} participantIds
  */
-export const updateDailyParticipants = async (pgClient, participants) => {
-  debug('Updating daily participants, count=%s', participants.size)
-  const ids = await mapParticipantsToIds(pgClient, participants)
+export const updateDailyParticipants = async (pgClient, participantIds) => {
+  debug('Updating daily participants, count=%s', participantIds.length)
   await pgClient.query(`
     INSERT INTO daily_participants (day, participant_id)
     SELECT now() as day, UNNEST($1::INT[]) AS participant_id
     ON CONFLICT DO NOTHING
   `, [
-    ids
+    participantIds
+  ])
+}
+
+/**
+ * @param {import('pg').Client} pgClient
+ * @param {import('./preprocess.js').Measurement[]} honestMeasurements
+ * @param {import('./preprocess.js').Measurement[]} allMeasurements
+ * @param {Map<string, number>} participantMap
+ * @param {Date} day
+ */
+export const updateStationAndParticipantDetails = async (
+  pgClient,
+  honestMeasurements,
+  allMeasurements,
+  participantMap,
+  day = new Date()
+) => {
+  const statsPerStationParticipant = new Map()
+  const participantSubnetsSet = new Set()
+
+  // JSON stringify the key to make it comparable, since objects compare by reference
+  /** @type {(m: import('./preprocess.js').Measurement, participantId: number) => string} */
+  const getStationParticipantKey = (m, participantId) => JSON.stringify({
+    stationId: m.stationId,
+    participantId
+  })
+  /** @type {(m: import('./preprocess.js').Measurement, participantId: number) => string} */
+  const getParticipantSubnetKey = (m, participantId) => JSON.stringify({
+    participantId,
+    subnet: m.inet_group
+  })
```

Comment:
These functions don't depend on the parent function's scope and can be moved out

Reply:
Ah I meant to send a message about this. This would presumably also apply to the existing internal function for updateDailyStations, right?

Reply:
it applies to all nested functions that don't need the parents' scope yeah

Reply:
Addressed in https://github.com/filecoin-station/spark-evaluate/pull/311/commits/e3845c9bd36cfddb719459fe72c3f2389b94ecda